File: 560_Subarray_Sum_Equals_K.py
class Solution:
    def subarraySum(self, nums, k):
        """
        :type nums: List[int]
        :type k: int
        :rtype: int
        """
        count, cur, res = {0: 1}, 0, 0
        for num in nums:
            cur += num
            res += count.get(cur - k, 0)
            count[cur] = count.get(cur, 0) + 1
        return res

File: 560_Subarray_Sum_Equals_K.cpp
class Solution {
public:
    int subarraySum(vector<int>& nums, int k) {
        unordered_map<int, int> dic;
        dic[0]++;
        int res = 0, cur = 0;
        for (int num : nums) {
            cur += num;
            res += dic[cur - k];
            dic[cur]++;
        }
        return res;
    }
};
